### DISCUSSION TITLE
Question about wallet utxo

### DISCUSSION THREAD

**Discussion by dxteam2024** [dogecoin/dogecoin#2884] [Help needed]
I'm developing a wallet and am a bit confused on the utxo model.
The UTXO for change is constructed locally, and the network has not yet confirmed the transaction.Can I submit  this unconfirmed change UTXO to the net? 
In testnet sometimes it works, sometimes it doesn't, I'm confused, does anyone know about this？

**Comment by patricklodder** [comment]
Any node that has the utxo in the mempool will accept a transaction spending it, to a configurable limit on both size and count, respectively 101kb and 25 ancestors or descendants. So as long as your transaction that created the utxo got accepted into the mempool of the node you're propagating to and does not exceed those or fee/dust limits, it should be accepted.

Do you have concrete examples of what didn't get accepted?

These are the parameters for accepting tx spending unconfirmed utxos:

```
  -limitancestorcount=<n>
       Do not accept transactions if number of in-mempool ancestors is <n> or
       more (default: 25)

  -limitancestorsize=<n>
       Do not accept transactions whose size with all in-mempool ancestors
       exceeds <n> kilobytes (default: 101)

  -limitdescendantcount=<n>
       Do not accept transactions if any ancestor would have <n> or more
       in-mempool descendants (default: 25)
```

**Reply by dxteam2024** [reply]
You mean whether unconfirmed UTXO can be accepted depends on the node's configuration on the above parameters?
I need to figure out what these parameters mean. And test the demo.
I rely on third-party node services, what kind of UTXO solution is the most suitable for wallet development？

**Reply by patricklodder** [reply]
What the above means is that if you're sending common transactions (i.e. under 1kb) you can do a series of 24 subsequent unconfirmed transactions until mempools stop accepting, assuming you pay enough fee and don't create dust outputs. It's safe to assume that over 90% of nodes have not changed those parameters, but it could help you in testing and figuring out if this is what obstructs you, if you run a testnet node.

When you say 3rd party nodes - how many do you connect to at a time?

Your strategy could simply be to (1) keep track of your unconfirmed transactions and (2) [re-broadcast unconfirmed transactions](https://github.com/dogecoin/dogecoin/blob/31afd133119dd2e15862d46530cb99424cf564b0/src/wallet/wallet.cpp#L1880) that are older than 5 blocks, because they ought to be mined by then, 99% of the time, currently.

**Reply by dxteam2024** [reply]
Thanks for your advice and patient, It helps me a lot.
